src/services/ScheduleService.ts

```diff
@@ -1,32 +1,53 @@
-import {Service} from "@tsed/di";
+import {Inject, Service} from "@tsed/di";
 import {Job, SimpleIntervalJob, ToadScheduler} from "toad-scheduler";
 import {SimpleIntervalSchedule} from "toad-scheduler/dist/lib/engines/simple-interval/SimpleIntervalSchedule";
 import {AsyncTask} from "toad-scheduler/dist/lib/common/AsyncTask";
+import schedule, {Job as DateJob, JobCallback} from "node-schedule";
+import {Logger} from "@tsed/logger";
+import {ObjectUtils} from "../utils/Utils";
 
 @Service()
 export class ScheduleService {
 
     private static readonly scheduler = new ToadScheduler();
 
-    public scheduleJob<T>(schedule: SimpleIntervalSchedule, jobHandler: (this: T) => Promise<void>, jobName: string, context: T): void {
+    private static readonly dateSchedules: DateJob[] = [];
+
+    @Inject()
+    private logger: Logger;
+
+    public scheduleJobInterval<T>(schedule: SimpleIntervalSchedule, jobHandler: (this: T) => Promise<void>, jobName: string, context: T): void {
         jobHandler = jobHandler.bind(context);
         const task = new AsyncTask(
             jobName,
             jobHandler
         );
-        const job = new SimpleIntervalJob(schedule, task);
+        const job = new SimpleIntervalJob(schedule, task, {
+            id: jobName
+        });
         ScheduleService.scheduler.addSimpleIntervalJob(job);
+        this.logger.info(`Registered interval job ${jobName}`);
     }
 
-    public stopJob(name: string): void {
-        ScheduleService.scheduler.stopById(name);
+    public scheduleJobAtDate<T>(name: string, when: Date, jobHandler: JobCallback, context: T): void {
+        if (when.getTime() < Date.now()) {
```

Comment:
I don't think this check is really needed but it doesn't hurt anything to keep.  Anything scheduled for a past date should just be immediately picked up.  But it also looks like the new submission round code prevents past dates anyway.

Reply:
> I don't think this check is really needed but it doesn't hurt anything to keep. Anything scheduled for a past date should just be immediately picked up. But it also looks like the new submission round code prevents past dates anyway.

only reason i kept this in is because this method is jsut a generic shedule date object and not really coupled to ending a round. it can be used to schedule anything (although, currently, it only is used by the deadline).  that is why i made this a service rather than putting it directly into the SubmissionRoundService if that makes sense